crates/autopilot/src/solvable_orders.rs

```diff
@@ -363,28 +335,25 @@ fn max_transfer_out_amount(order: &Order) -> Result<U256> {
 
 /// Keep updating the cache every N seconds or when an update notification happens.
 /// Exits when this becomes the only reference to the cache.
-async fn update_task(cache: Weak<SolvableOrdersCache>, current_block: CurrentBlockStream) {
+async fn update_task(
+    cache: Weak<SolvableOrdersCache>,
+    update_interval: Duration,
+    current_block: CurrentBlockStream,
+) {
     loop {
+        // We are not updating on block changes because
```

Comment:
We should still trigger updates on new blocks correct? Maybe this should be:
```suggestion
        // We are not exclusively updating on block changes because
```

Reply:
Before, the maintenance implementation triggered updates on new blocks. This was already not needed as the now deleted comment below here explains. Updating on block change isn't useful because we don't know if the event updater has already run and if the event updater hasn't run the auction wouldn't change because of new block state anyway.

Now, we have removed the notify channel and only update in fixed intervals (2 seconds).

Reply:
We could potentially intertwine the event updater and this module so that we can trigger updates right after the event updater has finished. But that only adds a small improvement to the average delay between new block and auction update as the current interval of 2 seconds is small compared to the expected block interval of 12.5 seconds. If we wanted to do that it would be a future PR.

Reply:
So, I agree that updating on every block isn't good enough, but I still think its useful.

IMO We should trigger updates to the solvable orders cache on every meaningful event (such as new blocks).